plugins/database/quota/src/org/apache/cloudstack/api/response/QuotaResponseBuilderImpl.java

```diff
@@ -0,0 +1,419 @@
+        response.setTariffValue(tariff.getCurrencyValue());
+        response.setEffectiveOn(tariff.getEffectiveOn());
+        response.setDescription(tariff.getDescription());
+        response.setCurrency(QuotaConfig.QuotaCurrencySymbol.value());
+        return response;
+    }
+
+    @Override
+    public QuotaBalanceResponse createQuotaBalanceResponse(List<QuotaBalanceVO> quotaBalance, Date startDate, Date endDate) {
+        if (quotaBalance.size() == 0) {
+            new InvalidParameterValueException("The request period does not contain balance entries.");
+        }
+        Collections.sort(quotaBalance, new Comparator<QuotaBalanceVO>() {
+            public int compare(QuotaBalanceVO o1, QuotaBalanceVO o2) {
+                return o2.getUpdatedOn().compareTo(o1.getUpdatedOn()); // desc
+            }
+        });
+
+        int quota_activity = quotaBalance.size();
+        QuotaBalanceResponse resp = new QuotaBalanceResponse();
+        BigDecimal lastCredits = new BigDecimal(0);
+        boolean consecutive = true;
+        for (Iterator<QuotaBalanceVO> it = quotaBalance.iterator(); it.hasNext();) {
+            QuotaBalanceVO entry = it.next();
+            s_logger.info("createQuotaBalanceResponse: Date=" + entry.getUpdatedOn().toGMTString() + " balance=" + entry.getCreditBalance() + " credit=" + entry.getCreditsId());
+            if (entry.getCreditsId() > 0) {
+                if (consecutive) {
+                    lastCredits = lastCredits.add(entry.getCreditBalance());
+                }
+                resp.addCredits(entry);
+                it.remove();
+            } else {
+                consecutive = false;
+            }
+        }
+
+        if (quota_activity > 0 && quotaBalance.size() > 0) {
+            // order is desc last item is the start item
+            QuotaBalanceVO startItem = quotaBalance.get(quotaBalance.size() - 1);
+            QuotaBalanceVO endItem = quotaBalance.get(0);
+            resp.setStartDate(startDate);
+            resp.setStartQuota(startItem.getCreditBalance());
+            resp.setEndDate(endDate);
+            resp.setEndQuota(endItem.getCreditBalance().add(lastCredits));
+        } else if (quota_activity > 0) {
+            // order is desc last item is the start item
+            resp.setStartDate(startDate);
+            resp.setStartQuota(new BigDecimal(0));
+            resp.setEndDate(endDate);
+            resp.setEndQuota(new BigDecimal(0).add(lastCredits));
+        } else {
+            resp.setStartQuota(new BigDecimal(0));
+            resp.setEndQuota(new BigDecimal(0));
+        }
+        resp.setCurrency(QuotaConfig.QuotaCurrencySymbol.value());
+        resp.setObjectName("balance");
+        return resp;
+    }
+
+    @Override
+    public QuotaStatementResponse createQuotaStatementResponse(final List<QuotaUsageVO> quotaUsage) {
+        if (quotaUsage == null || quotaUsage.size() == 0) {
```

Comment:
``checkArgument`` provides a consistent way to check arguments (e.g. always throwing ``IllegalArgumentException``) and allows the check to be one line rather than a 3 line if block.  Given the number of argument checks typically required, ``checkArgument`` can significantly improve code readability.